pkg/cmd/admin/prune/images.go

```diff
@@ -513,3 +768,312 @@ func getClients(f *clientcmd.Factory, caBundle string) (*client.Client, kclients
+	}
+
+	return podList, nil
+}
+
+func collectRegistryDeploymentModels(
+	dcNamespacer client.DeploymentConfigsNamespacer,
+	deploymentGetter kextclientset.DeploymentsGetter,
+	dsGetter kextclientset.DaemonSetsGetter,
+) ([]runtime.Object, error) {
+	var result []runtime.Object
+
+	dcList, err := dcNamespacer.DeploymentConfigs(DefaultRegistryNamespace).List(metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{kapi.ObjectNameField: "docker-registry"}).String(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	for i := range dcList.Items {
+		dc := &dcList.Items[i]
+		result = append(result, dc)
+	}
+
+	deploymentList, err := deploymentGetter.Deployments(DefaultRegistryNamespace).List(metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{kapi.ObjectNameField: "docker-registry"}).String(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	for i := range deploymentList.Items {
+		deployment := &deploymentList.Items[i]
+		result = append(result, deployment)
+	}
+
+	dsList, err := dsGetter.DaemonSets(DefaultRegistryNamespace).List(metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{kapi.ObjectNameField: "docker-registry"}).String(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	for i := range dsList.Items {
+		ds := &dsList.Items[i]
+		result = append(result, ds)
+	}
+
+	if len(result) == 0 {
+		return nil, fmt.Errorf("no deploymentconfig, daemonset nor deployment named docker-registry found in the default namespace")
+	}
+
+	return result, nil
+}
+
+func getMatchingPods(decoder runtime.Decoder, deploymentModel runtime.Object, podList *kapi.PodList) kapi.PodList {
+	result := *podList
+	result.Items = []kapi.Pod{}
+
+	for i := range podList.Items {
+		pod := &podList.Items[i]
+		sref := kapi.SerializedReference{}
+		annotation := pod.Annotations[kapi.CreatedByAnnotation]
+		err := runtime.DecodeInto(decoder, []byte(annotation), &sref)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "failed to decode %s annotation %q of pod %s", kapi.CreatedByAnnotation, annotation, pod.Name)
+			continue
+		}
+
+		dmMeta := deploymentModel.(metav1.ObjectMetaAccessor).GetObjectMeta()
+		if sref.Reference.Kind == deploymentModel.GetObjectKind().GroupVersionKind().Kind &&
+			sref.Reference.Namespace == dmMeta.GetNamespace() &&
+			sref.Reference.Name == dmMeta.GetName() &&
+			sref.Reference.UID == dmMeta.GetUID() {
+			result.Items = append(result.Items, *pod)
+		}
+	}
+
+	return result
+}
+
+func isDeploymentSuccessfullyDone(
+	osClient client.Interface,
+	kClient kclientset.Interface,
+	dm runtime.Object,
+) (bool, error) {
+	dmMeta := dm.(metav1.ObjectMetaAccessor).GetObjectMeta()
+	//m.GetNamespace(), m.GetName())
+	var statusViewer kubectl.StatusViewer
+	var err error
+
+	statusViewer, err = getStatusViewerForDeploymentModel(osClient, kClient, dm)
+	if err != nil {
+		return false, err
+	}
+
+	description, done, err := statusViewer.Status(dmMeta.GetNamespace(), dmMeta.GetName(), 0)
```

Comment:
@kargakis can I safely assume the deployment/daemonset/deploymentconfig has been successfully redeployed (at the time of the check) as long as `done==true && err==nil`?